EED_Promotions.module.php

```diff
@@ -1063,75 +1050,76 @@ public function get_applicable_items(
     }
 
 
-
     /**
      * generate_promotion_line_items
      * if the promotion in question has not already been redeemed for the given line item,
      * then have a line item generated by the promotion scope object, and increment
      *
+     * @param EE_Cart $cart
+     * @return    EE_Event[]
+     * @throws EE_Error
+     * @throws ReflectionException
      * @since     1.0.4
-     * @access    public
-     * @param    \EE_Cart $cart
-     * @return    \EE_Event[]
      */
-    public function get_events_from_cart(EE_Cart $cart)
+    public function get_events_from_cart(EE_Cart $cart): array
     {
         $event_line_items = $object_type_line_items = EEH_Line_Item::get_event_subtotals($cart->get_grand_total());
-        $events = array();
+        $events           = [];
         foreach ($event_line_items as $event_line_item) {
             if ($event_line_item instanceof EE_Line_Item) {
-                $events[ $event_line_item->OBJ_ID() ] = $event_line_item->get_object();
+                $events[$event_line_item->OBJ_ID()] = $event_line_item->get_object();
             }
         }
         return $events;
     }
 
 
-
     /**
      * Get events from a specific Transaction
      *
+     * @param EE_Transaction $transaction
+     * @return    EE_Event[]
+     * @throws EE_Error
+     * @throws ReflectionException
      * @since     1.0.4
-     * @access    public
-     * @param     \EE_Transaction $transaction
-     * @return    \EE_Event[]
      */
-    public function getEventsFromTransaction(EE_Transaction $transaction)
+    public function getEventsFromTransaction(EE_Transaction $transaction): array
     {
-        $event_line_items = $object_type_line_items = EEH_Line_Item::get_event_subtotals($transaction->total_line_item());
-        $events = array();
+        $event_line_items =
+        $object_type_line_items = EEH_Line_Item::get_event_subtotals($transaction->total_line_item());
+        $events           = [];
```

Comment:
This looks really odd. This would be better imo:

```suggestion
        $event_line_items = $object_type_line_items = EEH_Line_Item::get_event_subtotals(
            $transaction->total_line_item()
        );
        $events = [];
```